janus_core/cli/utils.py

```
@@ -306,3 +309,66 @@ def check_config(ctx: Context) -> None:
+    kwargs
+        CLI correlation keyword options.
+
+    Returns
+    -------
+    list[dict]
+        The parsed correlation_kwargs for md.
+    """
+    parsed_kwargs = []
+    for name, cli_kwargs in kwargs.value.items():
+        if "a" not in cli_kwargs and "b" not in cli_kwargs:
+            raise ValueError("At least one observable must be supplied as 'a' or 'b'")
+
+        if "points" not in cli_kwargs:
+            raise ValueError("Correlation keyword argument points must be specified")
+
+        # Accept an Observable to be replicated.
+        if "b" not in cli_kwargs:
+            a = cli_kwargs["a"]
+            b = deepcopy(a)
+            # Copying Observable, so can copy kwargs as well.
+            if "b_kwargs" not in cli_kwargs and "a_kwargs" in cli_kwargs:
+                cli_kwargs["b_kwargs"] = cli_kwargs["a_kwargs"]
+        elif "a" not in cli_kwargs:
+            b = cli_kwargs["b"]
+            a = deepcopy(b)
+            if "a_kwargs" not in cli_kwargs and "b_kwargs" in cli_kwargs:
+                cli_kwargs["a_kwargs"] = cli_kwargs["b_kwargs"]
+        else:
+            a = cli_kwargs["a"]
+            b = cli_kwargs["b"]
+
+        a_kwargs = cli_kwargs["a_kwargs"] if "a_kwargs" in cli_kwargs else {}
+        b_kwargs = cli_kwargs["b_kwargs"] if "b_kwargs" in cli_kwargs else {}
+
+        # Accept "." in place of one kwargs to repeat.
+        if a_kwargs == "." and b_kwargs == ".":
+            raise ValueError("a_kwargs and b_kwargs cannot 'ditto' each other")
+        if a_kwargs and b_kwargs == ".":
+            b_kwargs = a_kwargs
+        elif b_kwargs and a_kwargs == ".":
+            a_kwargs = b_kwargs
+
+        cor_kwargs = {
+            "name": name,
+            "points": cli_kwargs["points"],
+            "a": getattr(observables, a)(**a_kwargs),
+            "b": getattr(observables, b)(**b_kwargs),
+        }
+
+        for optional in cli_kwargs.keys() & {"blocks", "averaging", "update_frequency"}:
```

Do we want to raise an error if any keys aren't in this set?

---

Depends how you want defaults to be handled. Currently these have defaults in the correlator. They could be added here if not present instead I suppose.

Otherwise all those options will need to be specified on the cli which becomes long, I suppose it already is though. 

If that's the preference then these optional values can be required and an error raised.

---

I'm happy with defaults to be set wherever. The question was more than this looks like it silently throws away `cli_kwargs` keywords if they're not in `{"blocks", "averaging", "update_frequency"}`, unless I'm misunderstanding, which might be confusing.

---

I'd be inclined to raise as error, unless there's a reason not to? Otherwise you end up running an entire simulation  while not calculating what you actually wanted.

---

Yeah its raising an error now e.g. in tests for typo: 

```python
        (
            {"vaf": {"points": 10, "a": "Velocity", "a_kwarg": {}}},
            "correlation_kwargs got unexpected argument",
        ),
```

---

Ah was looking for it nearer here, nice, thanks!